rollpe/heart/apis.py

```diff
@@ -82,31 +97,51 @@ def post(self, request):
             비공개면 초대받지 못한 유저의 경우 작성할 수 없도록 제어 필요
         """
         
+        # 로그인 여부 검증
         if not request.user.is_authenticated:
             return Response(status=401)
         
-        
+        # serialzier 검증
         serializer = HeartWriteSerializer(data=request.data, method='post')
         if not serializer.is_valid():
-            return Response(status=400)
-        heart_instance = serializer.create(serializer.validated_data)
+            return Response(status=480)
+        
+        validated_data = serializer.validated_data
+        
+        # 존재하는 Paper인지 검증
+        try:
+            paper_instance = Paper.objects.get(pk=validated_data['paperFK'])
+        except Paper.DoesNotExist:
+            return Response(status=404)
+        
+        # 중복 작성 검증
+        if Heart.objects.filter(userFK=request.user.id, paperFK=paper_instance.id).exists():
+            return Response(status=482)
+        
+        # 비공개 롤링페이퍼일 경우만 검증
+        if not paper_instance.viewStat and not is_only_invited_user(request.user, paper_instance):
+            return Response(status=471)
+        
+        validated_data['userFK'] = request.user.id
+        heart_instance = serializer.create(validated_data)
+        
         return Response(status=201)
     
     
-    
     def patch(self, request):
         """
             본인이 작성한 게시물이 맞는지, 초대받은 유저가 맞는지 검증 필요
             로그인 구현되면 코드 추가할 것
```

Comment:
TODO를 추가해서 관리하는 방법도 좋아요

```suggestion
            TODO 로그인 구현되면 코드 추가할 것
```

Reply:
좋아연!!!!!